File: mobile/modules/audio-converter/ios/AudioConverterModule.swift
import ExpoModulesCore

public final class AudioConverterModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AudioConverter")

    AsyncFunction("prepareAudioAsync") { (inputURI: String, outputURI: String) -> [String: Any] in
      let inputURL = try AudioPreparer.localFileURL(from: inputURI)
      let outputURL = try AudioPreparer.localFileURL(from: outputURI)

      try AudioPreparer.prepare(inputURL: inputURL, outputURL: outputURL)

      return [
        "uri": outputURL.absoluteString,
        "temporary": true,
      ]
    }
  }
}

File: mobile/modules/audio-converter/src/AudioConverterModule.ts
import { NativeModule, requireNativeModule } from "expo";

export type PreparedAudio = {
  uri: string;
  temporary: boolean;
};

declare class AudioConverterModule extends NativeModule {
  prepareAudioAsync(inputUri: string, outputUri: string): Promise<PreparedAudio>;
}

export default requireNativeModule<AudioConverterModule>("AudioConverter");
